src/TheMovie/TheMovie/ViewModels/UpcomingMovieViewModel.cs

```diff
@@ -0,0 +1,198 @@
+            get => _isLoading;
+            set => SetProperty(ref _isLoading, value);
+        }
+
+        private bool _isLoadingUpcomingMovies;
+        public bool IsLoadingUpcomingMovies
+        {
+            get => _isLoadingUpcomingMovies;
+            set => SetProperty(ref _isLoadingUpcomingMovies, value);
+        }
+
+        private string _searchPlaceHolder;
+        public string SearchPlaceHolder
+        {
+            get => _searchPlaceHolder;
+            set => SetProperty(ref _searchPlaceHolder, value);
+        }
+
+        private string _searchText;
+        public string SearchText
+        {
+            get => _searchText;
+            set => SetProperty(ref _searchText, value);
+        }
+
+        public ObservableCollection<UpcomingMovieModel> UpcomingMovies { get; set; }
+
+        #endregion
+
+        #region Commands
+
+        public Command SearchCommand { get; set; }
+        public Command UpcomingMovieRefreshCommand { get; set; }
+        public DelegateCommand<UpcomingMovieModel> ItemAppearingCommand { get; set; }
+        public DelegateCommand<UpcomingMovieModel> ItemTappedCommand { get; set; }
+
+        #endregion
+
+        #region Services
+
+        private readonly IMovieService _movieService;
+        private readonly IGenreService _genreService;
+
+        #endregion
+
+        public UpcomingMovieViewModel(INavigationService navigationService, IPageDialogService pageDialogService, IMovieService movieService, IGenreService genreService)
+            : base(navigationService, pageDialogService)
+        {
+            _movieService = movieService;
+            _genreService = genreService;
+            Title = "Upcoming Movies";
+            SearchPlaceHolder = "Search your movie :)";
+            UpcomingMovies = new ObservableCollection<UpcomingMovieModel>();
+            SearchCommand = new Command(async () => await ExecuteSearchCommand());
+            UpcomingMovieRefreshCommand = new Command(async () => await ExecuteUpcomingMovieRefreshCommand());
+            ItemAppearingCommand = new DelegateCommand<UpcomingMovieModel>(ExecuteItemAppearingCommand);
+            ItemTappedCommand = new DelegateCommand<UpcomingMovieModel>(ExecuteItemTappedCommand);
+            InitializeUpcomingMovies();
+        }
+
+        #region Methods 
+
+        private async Task GetUpcomingMovies(int pageMovie = 1)
+        {
+            var movies = await _movieService.GetAll(pageMovie);
+
+            if (movies.UpcomingMovies.Count() > 0)
+            {
+                _pageUpcomingMovies = movies.Page;
+                _totalPagesUpcomingMovies = movies.TotalPages;
+
+                var genres = await _genreService.GetAll();
+
+                foreach (var movie in movies.UpcomingMovies)
+                {
+                    foreach (var genreId in movie.GenreIds)
+                    {
+                        movie.Genres += $"{genres.GenreMovies.FirstOrDefault(x => x.Id == genreId).Name}\n";
```

Comment:
By the way using of a string concatenation practice is not recommended to use in a loop or if you need to create a long string from a lot of smaller pieces. What alternatives can you offer?

Reply:
Put the data of Genres (string's) an array, then list on the screen. I think that should be a good solution to store the Genres in a local database (SQlite or Realm) because they do not change and avoiding an API call. If a new one appears Genres, register at this local database.